src/Control/SimpleSuggestions.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+namespace ipl\Web\Control;
+
+use Generator;
+use ipl\Html\Attributes;
+use ipl\Html\BaseHtmlElement;
+use ipl\Html\FormattedString;
+use ipl\Html\FormElement\ButtonElement;
+use ipl\Html\FormElement\InputElement;
+use ipl\Html\HtmlElement;
+use ipl\Html\Text;
+use Psr\Http\Message\ServerRequestInterface;
+use Traversable;
+
+abstract class SimpleSuggestions extends BaseHtmlElement
+{
+    /** @var int suggestions limit */
+    const DEFAULT_LIMIT = 10;
+
+    protected $tag = 'ul';
+
+    /** @var string The given input for search */
+    protected $searchTerm;
+
+    /** @var Traversable Fetched data for given input */
+    protected $data;
+
+    /** @var string Default first suggestion in the suggestion list */
+    protected $default;
+
+    /**
+     * Set the search term
+     *
+     * @param string $searchTerm
+     *
+     * @return $this
+     */
+    public function setSearchTerm(string $searchTerm): self
+    {
+        $this->searchTerm = $searchTerm;
+
+        return $this;
+    }
+
+    /**
+     * Set the fetched data
+     *
+     * @param $data Generator
+     *
+     * @return $this
+     */
+    public function setData(Generator $data): self
```

Comment:
`fetchSuggestions` returns `mixed`. This is incompatible. But I'm not a fan of requiring `Generator` here anyway. Remove it.